pylint/checkers/variables.py

```diff
@@ -2666,17 +2733,27 @@ def _check_unpacking(
         values = self._nodes_to_unpack(inferred)
         if values is not None:
             if len(targets) != len(values):
-                # Check if we have starred nodes.
-                if any(isinstance(target, nodes.Starred) for target in targets):
-                    return
-                self.add_message(
-                    "unbalanced-tuple-unpacking",
-                    node=node,
-                    args=(
+                if isinstance(inferred, DICT_TYPES):
+                    symbol = "unbalanced-dict-unpacking"
+                    args = (
+                        node.value.as_string(),
+                        len(targets),
+                        "s" if len(targets) > 1 else "",
+                        len(values),
+                        "s" if len(values) > 1 else "",
+                    )
+                else:
+                    symbol = "unbalanced-tuple-unpacking"
+                    args = (
```

Comment:
There's a mypy error that we need to fix here, probably because we added the pluralization..